ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/LedgerFactory.scala

```diff
@@ -15,16 +16,16 @@ import scopt.OptionParser
 
 import scala.concurrent.ExecutionContext
 
-trait LedgerFactory[T <: KeyValueLedger, ExtraConfig] {
+trait LedgerFactory[RWS <: ReadWriterService, ExtraConfig] {
   val defaultExtraConfig: ExtraConfig
 
   def extraConfigParser(parser: OptionParser[Config[ExtraConfig]]): Unit
 
-  def owner(config: Config[ExtraConfig], participantConfig: ParticipantConfig)(
+  def readWriterServiceOwner(config: Config[ExtraConfig], participantConfig: ParticipantConfig)(
```

Comment:
As a more general remark: what do you think about splitting the factory into `readServiceOwner` and `writeServiceOwner`? There are KV Ledger implementations where you don't want to create a read service when you only want to use a write service.

Reply:
It sounds sensible to me, trying to shape that.

Reply:
I've split the hierarchy upwards to clarify that a `LedgerFactory` is really a reader factory plus a writer factory plus a configuration builder: non-full implementations can really pick and choose in this way.

I'd leave `LedgerFactory` and `Runner` as they are though, in order not to force splitting present full ledger implementations into two halves: a full ledger has resource mgmt. options that separate reader and writers don't have (e.g. share the DB connection pool).